scripts/generate-comments.js

```diff
@@ -0,0 +1,171 @@
+    if (!('property_descriptions' in rule)) {
+        throw new Error("rule missing property_description:" + JSON.stringify(rule));
+    }
+    if (!('path_patterns' in rule)) {
+        throw new Error("rule missing path_patterns:" + JSON.stringify(rule));
+    }
+    maybeDebugLog("\nRule for type: " + type);
+    maybeDebugLog("  property_descriptions:")
+    Object.entries(rule.property_descriptions)
+        .forEach(([property, description]) => maybeDebugLog("    " +  property + ": " + description));
+    maybeDebugLog("  path_patterns: \n" + rule.path_patterns.map(entry => "  - " + toRegex(entry)).join("\n"));
+})
+
+// Read in the input file
+const fileContent = fs.readFileSync(inputFile, "utf-8");
+const fileDoc = yaml.parseDocument(fileContent);
+let counter = 0;
+let lastNode = null;
+// Visit each key/value pair in the input file YAML, attempting to match against description rules
+// and setting a comment with the description from the matching rule
+yaml.visit(fileDoc, {
+    Pair: (key, node, path) => {
+        if (yaml.isSeq(node.value)) {
+            node.value.items.forEach(item => item.commentBefore = null);
+        }
+        let prevLastNode = lastNode;
+        lastNode = node;
+        counter++;
+        // Compute the parenPath, a string representation of the location of this key/value pair in the document
+        // For example, the following sample YAML is annotated with the parentPath at each node
+        // parent:                      # .
+        //   child: value               # .parent
+        //   child_arr:                 # .parent
+        //     - arr_key: value         # .parent.child_arr[]
+        const parentPath = pathToString(path);
+        const propertyKey = node.key.value;
+        maybeDebugLog("");
+        maybeDebugLog("parentPath: " + parentPath );
+        maybeDebugLog("propertyKey: " + propertyKey);
+        maybeDebugLog("currentNodePath: " + parentPath + (parentPath === "." ? "" : ".") + propertyKey);
+        // Iterate through the rules and find the first with a matching entry in rule.path_patterns
+        const matchingRule = typeDescriptionsYaml.find((rule) => {
+            const matchingPathPattern = rule['path_patterns'].find((pathPattern) => {
+                const regex = new RegExp(toRegex(pathPattern));
+                return regex.test(parentPath);
+            });
+            return matchingPathPattern !== undefined;
+        });
+        // Exit early if no matching rule
+        if (matchingRule === undefined) {
+            maybeDebugLog("no matching rule")
+            return;
+        }
+        maybeDebugLog("matched rule: " + matchingRule.type);
+        // Check if there is a description for the current propertyKey in the matching rule
+        // Exit early if none registered
+        const description = matchingRule['property_descriptions'][propertyKey];
+        if (description === undefined) {
+            maybeDebugLog("no matching property")
```

Comment:
is this an error - do we want comments for all entries?

Reply:
Same [comment](https://github.com/open-telemetry/opentelemetry-configuration/pull/104#discussion_r1721910849) applies here.